src/lib/y2storage/proposal_settings.rb

```diff
@@ -84,32 +137,74 @@ class ProposalUserSettings
     #   * :all Delete all Windows partitions, even if not needed.
     #
     #   @raise ArgumentError if any other value is assigned
-    attr_accessor :windows_delete_mode
+    attr_reader :windows_delete_mode
 
+    # @note :legacy and :ng formats
     # @return [Symbol] what to do regarding removal of existing Linux
     #   partitions. See {DiskAnalyzer} for the definition of "Linux partitions".
     #   @see #windows_delete_mode for the possible values and exceptions
-    attr_accessor :linux_delete_mode
+    attr_reader :linux_delete_mode
 
+    # @note :legacy and :ng formats
     # @return [Symbol] what to do regarding removal of existing partitions that
     #   don't fit in #windows_delete_mode or #linux_delete_mode.
     #   @see #windows_delete_mode for the possible values and exceptions
-    attr_accessor :other_delete_mode
-
-    def initialize
-      @use_lvm                  = false
-      self.encryption_password  = nil
-      @root_filesystem_type     = Filesystems::Type::BTRFS
-      @use_snapshots            = true
-      @use_separate_home        = true
-      @home_filesystem_type     = Filesystems::Type::XFS
-      @enlarge_swap_for_suspend = false
-      @resize_windows           = true
-      @windows_delete_mode      = :ondemand
-      @linux_delete_mode        = :ondemand
-      @other_delete_mode        = :ondemand
+    attr_reader :other_delete_mode
+
+    # @note :ng format
+    # @return [Symbol] if the user decides to use LVM, strategy to decide the size
+    #   of the volume group (and, thus, the number and size of created physical volumes).
+    #
+    #   * :use_available The VG will be created to use all the available space, thus the
```

Comment:
NP: Details in the yardoc formatting:

1)  `:use_available` vs `use_needed:` inconsistent placing of the colon.
2) For long descriptions that are not a one-liner we usually do

```ruby
# Description
#
# @note bla
#
# @return [Symbol]
attr_reader :whatever
```

Instead of that syntax with the `@return` on top.